### ISSUE TITLE
Incident Timeline

### ISSUE THREAD

**Issue by matejkramny** [cachethq/Cachet#530]
I'd like to start a discussion on how incidents are displayed..

At the moment, if an incident is created, and then 5 hours later resolved, it shows as '5 hours ago', where it would be useful to see when it was resolved..

Example:

```
- Frontend Down - Resolved now
  update message
- Backend down - Created 5 hours ago
  message...
- Frontend Down - Created 5 hours ago
  message..
```

**Comment by jbrooksuk** [comment]
The problem we currently have is that incidents aren't linked together. This will change in V2 though.

**Comment by jbrooksuk** [comment]
Essentially we have no way of knowing when an incident was resolved. Plus we don't have any constraints on only adding an incident which affects a component (which would make much more sense, but be restrictive).

**Comment by SamuelMoraesF** [comment]
Maybe something like?!

![captura de tela de 2015-04-02 21 54 50](https://cloud.githubusercontent.com/assets/3705263/6976629/fae1ceda-d982-11e4-9c44-1b7e0241ee31.png)

**Comment by matejkramny** [comment]
Exactly!

**Comment by jbrooksuk** [comment]
@SamuelMoraesF I like that! It's probably the kind of thing we'll be doing for V2.

**Comment by jbrooksuk** [comment]
FYI, I want to get this done, but I can't until V2 is being worked on due to the limitations on the existing incidents system.

**Comment by jbrooksuk** [comment]
Just a quick update on this issue:

We'll be doing this for Version 2. In V2 incidents and components will become self-contained entities. By this, I mean that you'll be able to browse the history of a component for example at `https://status.cachethq.io/component/1` or `https://status.cachethq.io/component/api` and it'll show all of the incidents that have affected the component.

Not only that, but like @SamuelMoraesF showed above, incidents will change too. Imagine it like a forum, when you create an incident you're creating a new topic. As that incident develops you'll be adding a reply (an update) which may or may not also change the status.

This change alone will redefine how Cachet is used, whilst hopefully keeping backwards compatible, so if you want to, you could just create a bunch of incidents with the status changing as we do now.

Once this is done we'll also be able to close:

- #547 
- #236

Which will be awesome.

**Comment by ryanwalls** [comment]
Has this been resolved yet with the v2 beta releases?

**Comment by jbrooksuk** [comment]
Unfortunately not. This requires a pretty big rewrite to:

1. The status page itself.
2. The dashboard.
3. The API.

It's something we're planning for V3, since V2 is just around the corner.

**Comment by jbrooksuk** [comment]
Merged into https://github.com/cachethq/Cachet/issues/1176
